src/Sulu/Bundle/WebsiteBundle/EventListener/AppendAnalyticsListener.php

```diff
@@ -82,16 +88,55 @@ public function onResponse(FilterResponseEvent $event)
         }
 
         $portalUrl = $this->requestAnalyzer->getAttribute('urlExpression');
-        $analytics = $this->analyticsRepository->findByUrl(
+        /** @var Analytics[] $analyticsArray */
+        $analyticsArray = $this->analyticsRepository->findByUrl(
             $portalUrl,
             $this->requestAnalyzer->getPortalInformation()->getWebspaceKey(),
             $this->environment
         );
 
-        $content = $this->engine->render('SuluWebsiteBundle:Analytics:website.html.twig', ['analytics' => $analytics]);
+        $analyticsContent = [
+            self::POSITION_HEAD_OPEN => null,
+            self::POSITION_HEAD_CLOSE => null,
+            self::POSITION_BODY_OPEN => null,
+            self::POSITION_BODY_CLOSE => null,
+        ];
+        foreach ($analyticsArray as $analytics) {
+            $type = $analytics->getType();
+            foreach ($analyticsContent as $tag => &$value) {
+                $template = 'SuluWebsiteBundle:Analytics:' . $type . DIRECTORY_SEPARATOR . $tag . '.html.twig';
+
+                if (!$this->engine->exists($template)) {
+                    continue;
+                }
+
+                $value .= $this->engine->render($template, ['analytics' => $analytics]);
+            }
+        }
 
         $response = $event->getResponse();
-        $responseContent = $response->getContent();
-        $response->setContent(str_replace('</body>', $content . '</body>', $responseContent));
+        $content = $response->getContent();
+
+        if ($analyticsContent[self::POSITION_HEAD_OPEN]) {
+            $matches = [];
+            preg_match('/<head[^>]*>/', $content, $matches);
```

Comment:
could be extracted into a function. set the regex into a static variable and iterate over it.

Reply:
something like this:

```php
foreach (self::POSITIONS as $position => $regex) {
	$analyticsContent = $this->getAnalyticsContent($analyticsArray, $position);

    preg_match($regex, $content, $matches);
    $content = str_replace($matches[0],  $matches[0] . $analyticsContent, $content);
}
```